### ISSUE TITLE
Bad (Incomplete) "DependsOn" when updating an AppSync GraphQL Schema to be searchable

### ISSUE THREAD

**Issue by nagey** [aws-amplify/amplify-cli#591]
**Describe the bug**
Updating a model to `@searchable @versioned` on an existing AppSync API fails (looks like due to a race condition) 

**To Reproduce**
1. Setup an API with models
2. Add a Model
3. Make the model searchable (in your `schema.graphql`
4. Watch amplify push (specifically the cloudformation build) fail

**Expected behavior**
"He fixes the cable?" — The Dude

**Screenshots**

Here is the (partial) log of the `amplify push` command:
```
⠋ Updating resources in the cloud. This may take a few minutes...

UPDATE_IN_PROGRESS capbase-20181117083404 AWS::CloudFormation::Stack Sun Dec 16 2018 15:23:19 GMT-0500 (EST) User Initiated
UPDATE_IN_PROGRESS analyticscapbase       AWS::CloudFormation::Stack Sun Dec 16 2018 15:23:23 GMT-0500 (EST)               
UPDATE_IN_PROGRESS authcognitoab2d7a9c    AWS::CloudFormation::Stack Sun Dec 16 2018 15:23:23 GMT-0500 (EST)               
UPDATE_COMPLETE    analyticscapbase       AWS::CloudFormation::Stack Sun Dec 16 2018 15:23:24 GMT-0500 (EST)               
UPDATE_COMPLETE    authcognitoab2d7a9c    AWS::CloudFormation::Stack Sun Dec 16 2018 15:23:24 GMT-0500 (EST)               
UPDATE_IN_PROGRESS hostingS3AndCloudFront AWS::CloudFormation::Stack Sun Dec 16 2018 15:23:25 GMT-0500 (EST)               
UPDATE_COMPLETE    hostingS3AndCloudFront AWS::CloudFormation::Stack Sun Dec 16 2018 15:23:25 GMT-0500 (EST)               
UPDATE_IN_PROGRESS apicapbase             AWS::CloudFormation::Stack Sun Dec 16 2018 15:23:27 GMT-0500 (EST)               
⠼ Updating resources in the cloud. This may take a few minutes...

UPDATE_IN_PROGRESS capbase-20181117083404-apicapbase-1TOT5NZ1RJIJD AWS::CloudFormation::Stack Sun Dec 16 2018 15:23:27 GMT-0500 (EST) User Initiated
⠙ Updating resources in the cloud. This may take a few minutes...

CREATE_IN_PROGRESS SearchContactInquiryResolver    AWS::AppSync::Resolver      Sun Dec 16 2018 15:23:37 GMT-0500 (EST)                            
CREATE_IN_PROGRESS MarketingBlogPostTable          AWS::DynamoDB::Table        Sun Dec 16 2018 15:23:37 GMT-0500 (EST)                            
CREATE_IN_PROGRESS SearchMarketingBlogPostResolver AWS::AppSync::Resolver      Sun Dec 16 2018 15:23:37 GMT-0500 (EST)                            
CREATE_IN_PROGRESS ContactInquiryTable             AWS::DynamoDB::Table        Sun Dec 16 2018 15:23:37 GMT-0500 (EST)                            
CREATE_IN_PROGRESS MarketingBlogPostTable          AWS::DynamoDB::Table        Sun Dec 16 2018 15:23:37 GMT-0500 (EST) Resource creation Initiated
CREATE_IN_PROGRESS ContactInquiryTable             AWS::DynamoDB::Table        Sun Dec 16 2018 15:23:38 GMT-0500 (EST) Resource creation Initiated
UPDATE_IN_PROGRESS GraphQLSchema                   AWS::AppSync::GraphQLSchema Sun Dec 16 2018 15:23:38 GMT-0500 (EST)                            
⠋ Updating resources in the cloud. This may take a few minutes...

CREATE_FAILED               SearchContactInquiryResolver                    AWS::AppSync::Resolver      Sun Dec 16 2018 15:23:39 GMT-0500 (EST) No field named searchContactInquirys found on type Query (Service: AWSAppSync; Status Code: 404; Error Code: NotFoundException; Request ID: 7917d29f-0170-11e9-8600-bf4bca63f89b)                                     
CREATE_FAILED               SearchMarketingBlogPostResolver                 AWS::AppSync::Resolver      Sun Dec 16 2018 15:23:40 GMT-0500 (EST) No field named searchMarketingBlogPosts found on type Query (Service: AWSAppSync; Status Code: 404; Error Code: NotFoundException; Request ID: 791a917a-0170-11e9-83e1-9f8f6a81e7b0)                                  
CREATE_FAILED               ContactInquiryTable                             AWS::DynamoDB::Table        Sun Dec 16 2018 15:23:40 GMT-0500 (EST) Resource creation cancelled                                                                                                                                                                                           
CREATE_FAILED               MarketingBlogPostTable                          AWS::DynamoDB::Table        Sun Dec 16 2018 15:23:40 GMT-0500 (EST) Resource creation cancelled                                                                                                                                                                                           
UPDATE_FAILED               GraphQLSchema                                   AWS::AppSync::GraphQLSchema Sun Dec 16 2018 15:23:41 GMT-0500 (EST) Resource update cancelled                                                                                                                                                   
```

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Terminal
 - Version 0.1.34 (from aws-amplify ^1.1.10)

**Smartphone (please complete the following information):**
N/A

**Additional context**
To me, this looks like a race condition / mishandled dependency in the auto-generated CF template. The `SearchContactInquiryResolver` that fails, `DependsOn` the ElasticSearchDataSource, but it should also be dependent on the update to the `GraphQLSchema` (which you can see has already started to update, but hasn't yet completed.

And, in fact, when I manually change the `DependsOn` field to include both of those items, the update works just fine.

**Comment by mikeparisstuff** [comment]
Addressed in #596. The code had the right intentions but was calling the method incorrectly.

**Comment by UnleashedMind** [comment]
The fix has been released last week, I'm going to close the issue, feel free to reopen it if needed.
